src/main/Serpiente.java

```diff
@@ -102,37 +102,35 @@ public Integer getId() {
 
 		@Override
 		public Estado moverse() {
-			for(int i = ubicaciones.size() - 1 ; i >= 0 ; i--) {
-				if (i != 0) {
+			for(int i = ubicaciones.size() - 1 ; i > 0 ; i--) {
 					ubicaciones.set(i, ubicaciones.get(i-1));
-				} else {
-					Ubicacion cabeza = ubicaciones.get(i);
-					int x = mirandoX != 0 ? cabeza.getX() + (mirandoX * velocidad) : cabeza.getX();
-					int y = mirandoY != 0 ? cabeza.getY() + (mirandoY * velocidad) : cabeza.getY();
+				}
+			Ubicacion cabeza = ubicaciones.get(0);
+			int x = cabeza.getX() + (mirandoX * velocidad);
+			int y = cabeza.getY() + (mirandoY * velocidad);
 					
-					ubicaciones.set(0, new Ubicacion(x,y));
-				}	
-			}
+			ubicaciones.set(0, new Ubicacion(x,y));
 			return this;
-		}
+		}	
+		
+
 
 		@Override
 		public Estado checkearColision(Serpiente serpiente) {
 			Ubicacion cabeza = ubicaciones.get(0);
 			List<Ubicacion> cuerpo = serpiente.getUbicaciones();
 			Integer idEnemigo = serpiente.getId();
-			for (int i = 0; i < cuerpo.size(); i++) {
+			if(cabeza.equals(serpiente.getUbicaciones().get(0)) && !soyYo(idEnemigo)) {
```

Comment:
Como hablamos con el profe ayer, en vez de el método negro (que yo hice) soyYo usa equals :v

Reply:
ah mal, igual no sé si no habría q cambiar todo ese método. Ahora me pongo a ver si me sale un mini-diseño de "evaluar()"

Reply:
pero por el momento estaba bueno que se mergeen los cambios :V pero como gustes

Reply:
vos decís que haga this.id.equals(idEnemigo) o this.equals(idEnemigo)?
Porque si lo que proponés es eso, no funciona, ya que el this hace referencia a la inner class, y no tiene un id.